Data/pubtator_to_xml.py

```diff
@@ -0,0 +1,161 @@
+import gzip
+import cgi
+import time
+import sys
+from lxml.builder import E
+from lxml.etree import tostring
+
+from bioc import BioCWriter, BioCCollection, BioCDocument, BioCPassage
+from bioc import BioCAnnotation, BioCLocation
+import tqdm
+
+
+def get_annotations(tag, index):
+    """Get Annotations
+
+    Keyword Arguments:
+    tag -- the annotation line that was parsed into an array
+    index -- the id of each document specific annotation
+    """
+
+    annt = BioCAnnotation()
+    annt.id = str(index)
+    annt.infons["type"] = tag[4]
+
+    # If the annotation type is a Gene,Species, Mutation, SNP
+    # Write out relevant tag
+    if tag[4] == "Gene":
+        annt.infons["NCBI Gene"] = tag[5]
+
+    elif tag[4] == "Species":
+        annt.infons["NCBI Species"] = tag[5]
+
+    elif "Mutation" in tag[4]:
+        annt.infons["tmVar"] = tag[5]
+
+    elif "SNP" in tag[4]:
+        annt.infons["tmVar"] = tag[5]
+
+    else:
+        # If there is no MESH ID for an annotation
+        if len(tag) > 5:
+            # check to see if there are multiple mesh tags
+            if "|" in tag[5]:
+                # Write out each MESH id as own tag
+                for tag_num, ids in enumerate(tag[5].split("|")):
+                    # Some ids dont have the MESH:#### form so added case to that
+                    if ":" not in ids:
+                        annt.infons["MESH {}".format(tag_num)] = tag[5]
+                    else:
+                        term_type, term_id = ids.split(":")
+                        annt.infons["{} {}".format(term_type, tag_num)] = term_id
+            else:
+                # Some ids dont have the MESH:#### form so added case to that
+                if ":" in tag[5]:
+                    term_type, term_id = tag[5].split(":")
+                    annt.infons[term_type] = term_id
+                else:
+                    annt.infons["MESH"] = tag[5]
+        else:
+            annt.infons["MESH"] = "Unknown"
+
+    location = BioCLocation()
+    location.offset = str(tag[1])
+    location.length = str(len(tag[3]))
+    annt.locations.append(location)
+    annt.text = tag[3]
+    return annt
+
+
+def convert_pubtator(input_file, output_file):
+    """Convert pubtators annotation list to XML
+
+    Keyword Arguments:
+    input_file -- the path of pubtators annotation file
+    output_file -- the path to output the converted text
+    """
+    # Set up BioCWriter to write specifically Pubtator
+    # Can change to incorporate other sources besides pubtator
+    writer = BioCWriter()
+    writer.collection = BioCCollection()
+    collection = writer.collection
+    collection.date = time.strftime("%Y/%m/%d")
+    collection.source = "Pubtator"
+    collection.key = "Pubtator.key"
+
+    with open(output_file, 'wb') as g:
```

Comment:
If we have encoding issues, we may need:

```
codecs.open(output_file, 'w', 'utf-8')
```

We can also consider `from eight import *` (https://github.com/kislyuk/eight). But I'm hesitant to recommend `eight` since I don't know how production ready it is. If it works correctly, it will make Python 2 behave like Python 3.